src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -380,6 +380,11 @@ public void RevertRow(long rowId)
             {
                 throw new ArgumentOutOfRangeException(nameof(rowId), SR.EditDataUpdateNotPending);
             }
+
+            if (typeof(RowCreate) == removedEdit.GetType())
```

Comment:
This seems like a great opportunity to add some tests - you should be able to set up a unit test that creates a session and then verifies that reverting the row creation results in the expected state (number of rows, row ID, etc.)